deployments/helm-chart/templates/controller-hpa.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if and .Values.controller.autoscaling.enabled (or (eq .Values.controller.kind "deployment")) -}}
+apiVersion: autoscaling/v2
```

Comment:
`autoscaling/v2` is only available starting from k8s 1.23, so I think we need to check the version before deploying this

Reply:
@lucacome Yes I believe you are correct. We need to verify this.

Reply:
That is why I left both types of resources. Wasn't sure how much templating we wanted to do to support older versions.

Reply:
what do you mean by both types?

Reply:
@coolbry95 can you clarify by both types?

Reply:
Types is the wrong word. I should have said apiVersion. I think this is the simplest way though. Just support autoscaling/v2.

For example we could template and add conditionals to support autoscaling/v1 and autoscaling/v2 at the same time.